src/plugins/validation/2and3/semantic-validators/responses.js

```diff
@@ -10,18 +10,93 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
   result.warning = [];
 
   config = config.responses;
-
   walk(jsSpec, [], function(obj, path) {
     const contentsOfResponsesObject = path[path.length - 1] === 'responses';
     const isRef = !!obj.$ref;
-
     if (contentsOfResponsesObject && !isRef) {
       each(obj, (response, responseKey) => {
         if (isOAS3) {
           each(response.content, (mediaType, mediaTypeKey) => {
-            const hasInlineSchema = mediaType.schema && !mediaType.schema.$ref;
-            if (
-              hasInlineSchema &&
+            if (mediaType && mediaType.schema && mediaType.schema.oneOf) {
+              for (let i = 0; i < mediaType.schema.oneOf.length; i++) {
+                const hasInlineSchema =
+                  mediaType.schema &&
+                  mediaType.schema.oneOf &&
```

Comment:
You already know that `mediaType.schema` and `mediaType.schema.oneOf` are `true`, these are redundant checks.